### PR TITLE
bulk-cdk: add feature flag environments

### PR DESCRIPTION
**Pull request by postamar**
## What

Fixes https://github.com/airbytehq/airbyte-internal-issues/issues/9704

The Bulk CDK needs to support the `DEPLOYMENT_MODE` environment variable and the associated `CLOUD` feature flag.
This PR adds support for this and makes source-mysql-v2 adopt it.

## How
As far as I can tell, none of the other feature flags from the legacy CDK matter for connectors.
In any case, the Bulk CDK now has a Micronaut Factory for injecting `Set<FeatureFlag>` where `FeatureFlag` is an enum and the injected value is the set of active feature flags. This is used in source-mysql-v2 to cause CHECK to fail when a config doesn't have enough encryption.

## Review guide
Commit by commit

## User Impact
None

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by edgao** [comment]
@johnny-schmidt you already did something for this I think? at least destination-dev-null respects cloud vs oss mode

**Comment by postamar** [comment]
Interesting! I searched and found it in the test fixtures, the substring to search for is `"deployment.mode"`. In my PR I hadn't thought of how this would impact the test runners, I'll have to figure something out. Back to the drawing board!

File: airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/command/FeatureFlag.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * An enum of all feature flags, currently these are set via environment vars.
+ *
+ * Micronaut can inject a Set<FeatureFlag> singleton of all active feature flags.
+ */
+enum class FeatureFlag(
+    val micronautEnvironmentName: String,
+    internal val envVar: EnvVar,
+    internal val requiredEnvVarValue: String,
```
Comment (edgao) [inline airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/command/FeatureFlag.kt:20]: fyi I'm hoping to merge https://github.com/airbytehq/airbyte/pull/46363 soon, which adds a `docker run` version of DestinationProcess

specifically this line (https://github.com/airbytehq/airbyte/pull/46363/files#diff-3e529f9238c6c09b63750cc36fab22aef78c72fe55172ff6011244a4c41fd39fR108) wants the actual env var value.

Which I think works out fine, it just means this field should be public?

Comment (postamar) [reply airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/command/FeatureFlag.kt:20]: It should work. Yes let's make all of this public.

Comment (postamar) [reply airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/command/FeatureFlag.kt]: You can now replace your line with `FeatureFlag.AIRBYTE_CLOUD_DEPLOYMENT.envVarBindingDeclaration`

**Comment by postamar** [comment]
This is ready for review, the only failing checks are connnector version bumps.

**Comment by johnny-schmidt** [comment]
> @johnny-schmidt you already did something for this I think? at least destination-dev-null respects cloud vs oss mode
@postamar @edgao 

Sorry, I missed this earlier. Micronaut automagically propagates env variables as properties. You get `"deployment.mode"` and any other env variable for free. The tests override the properties so that we can test both envs, but specifying them as properties isn't necessary in production.

I'm not sure what this PR accomplishes on top of that other than to limit the list of supported vars? However, you can also get that by specifying includes/excludes when building the ApplicationContext.

**Comment by johnny-schmidt** [comment]
https://docs.micronaut.io/latest/api/io/micronaut/context/ApplicationContextBuilder.html#environmentVariableIncludes(java.lang.String...)

**Comment by postamar** [comment]
> I'm not sure what this PR accomplishes on top of that other than to limit the list of supported vars?

Limiting the supported vars is not the goal per se but there is value in having a clear catalog of which env vars are meaningful. The main purpose of this PR is to facilitate activating feature flags in regular junit tests which are not `@MicronautTest`s. I understand that this may not be relevant to destinations.

**Review by johnny-schmidt — APPROVED** [review]
Pardon the delay. I had to confer with the team to make sure it wasn't breaking anything test-related

**Comment by postamar** [comment]
No worries! Like I said it's not urgent and I only just got the build to pass. Thanks for taking a look.

**Review by xiaohansong — APPROVED** [review]

**Pull request merged**